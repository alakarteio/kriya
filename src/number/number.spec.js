/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Number from './number'

const snap = (props) => {
  const component = renderer.create(
    <Number {...props} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Number', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should add children', () => snap({ children: 'a child' }))
    it('should have a default behaviour', () => snap({}))
    it('should adapt for a map', () => snap({ map: true }))
    it('should be secondary', () => snap({ secondary: true }))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
