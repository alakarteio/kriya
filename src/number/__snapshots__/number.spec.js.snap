// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Number graphical (JSX) should adapt for a map 1`] = `
<div
  className="number map primary"
  style={Object {}}
/>
`;

exports[`common/Number graphical (JSX) should add children 1`] = `
<div
  className="number primary"
  style={Object {}}
>
  a child
</div>
`;

exports[`common/Number graphical (JSX) should add custom className 1`] = `
<div
  className="number custom primary"
  style={Object {}}
/>
`;

exports[`common/Number graphical (JSX) should add custom style 1`] = `
<div
  className="number primary"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;

exports[`common/Number graphical (JSX) should be secondary 1`] = `
<div
  className="number secondary"
  style={Object {}}
/>
`;

exports[`common/Number graphical (JSX) should have a default behaviour 1`] = `
<div
  className="number primary"
  style={Object {}}
/>
`;
